app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala

```diff
@@ -185,9 +185,13 @@ class MovementsRepositoryImpl @Inject() (
 
     mongoRetry(Try(collection.aggregate[MovementWithoutMessages](aggregates)) match {
       case Success(obs) =>
-        obs.headOption().map {
-          opt => Right(opt)
-        }
+        obs
+          .head()
```

Comment:
Better to use `headOption` here still:

```scala
obs.headOption().map {
    case Some(x) => Right(x)
    case None    => Left(DocumentNotFound(s"No movement found with the given id: ${movementId.value}"))
}
```

I'm surprised `obs.head()` returns `null`, but we should still use `headOption()` to make it clear we might not get something back.